ext/search_kit/Civi/Api4/Action/SearchDisplay/SaveFile.php

```diff
@@ -0,0 +1,156 @@
+      // Remove buttons/menus and other column types that cannot be rendered in a spreadsheet
+      if (empty($col['key'])) {
+        unset($settings['columns'][$index]);
+      }
+      // Avoid wasting time processing links, editable and other non-printable items from spreadsheet
+      else {
+        \CRM_Utils_Array::remove($settings['columns'][$index], 'link', 'editable', 'icons', 'cssClass');
+      }
+    }
+    // Reset indexes as some items may have been removed
+    $settings['columns'] = array_values($settings['columns']);
+
+    // Displays are only exportable if they have actions enabled
+    if (empty($settings['actions'])) {
+      \CRM_Utils_System::permissionDenied();
+    }
+
+    // Force limit if the display has no pager
+    if (!isset($settings['pager']) && !empty($settings['limit'])) {
+      $apiParams['limit'] = $settings['limit'];
+    }
+    $apiParams['orderBy'] = $this->getOrderByFromSort();
+    $this->augmentSelectClause($apiParams, $settings);
+
+    $this->applyFilters();
+
+    $apiResult = civicrm_api4($entityName, 'get', $apiParams);
+
+    $rows = $this->formatResult($apiResult);
+
+    if ($this->appendDate) {
+      $this->fileName .= date("_Ymd", time());
+    }
+    $this->fileName .= '.' . ('array' === $this->format ? 'json' : $this->format);
+
+    // Download file contents.
+    if ('array' === $this->format) {
+      $content = $this->processData($this->display['label'], $this->display['settings']['columns'], $rows, $fileName);
+      $searchDisplayContent = json_encode($content);
+    }
+    else {
+      ob_start();
+      $this->processData($this->display['label'], $this->display['settings']['columns'], $rows, $fileName);
+      $searchDisplayContent = ob_get_contents();
+      ob_end_clean();
+    }
+
+    if (!empty($searchDisplayContent)) {
+      $config = \CRM_Core_Config::singleton();
+      $directoryName = $config->customFileUploadDir;
+      $fileName = \CRM_Utils_File::makeFileName($this->fileName);
+      $info = pathinfo($fileName);
+      // Regex in makeFileName blocks json due to check for js file names. Need to fix for saving as json.
+      if ('unknown' === $info['extension'] && 'array' === $this->format) {
+        $fileName = str_replace('unknown', 'json', $fileName);
+        $fileName = str_replace('_json', '', $fileName);
+      }
+
+      // Append folder name if provided.
+      if (!empty($this->folderName)) {
+        $folderName = preg_replace('/[-\s]/', '_', preg_replace('/[^\w\s_]/', '', $this->folderName));
+        // After replacement make sure we still have something for the folder name.
+        if (!empty($folderName)) {
+          $directoryName .= $folderName . '/';
+        }
+      }
+      \CRM_Utils_File::createDir($directoryName);
+      $fileParams = [
+        'name' => $this->fileName,
+        'mime_type' => $this->formats[$this->format]['mime'],
+        'uri' => $directoryName . $fileName,
+        'document' => $searchDisplayContent,
```

Comment:
`'document'` is deprecated. I don't think it's needed here.

https://github.com/civicrm/civicrm-core/blob/830bfa4d9c4672eba072ae387b89b78c17079da9/schema/Core/File.entityType.php#L49-L56

Reply:
@colemanw Removed the deprecated column.